cmd/kubectl-k8ssandra/register/register_test.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	if err := client2.Create((*multiEnv)[1].Context, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "dest-namespace"}}); err != nil {
+		t.Fatal(err)
+	}
+
+	buildDir := os.Getenv("BUILD_DIR")
+	if buildDir == "" {
+		_, b, _, _ := runtime.Caller(0)
+		buildDir = filepath.Join(filepath.Dir(b), "../../../build")
+	}
+
+	if _, err := os.Stat(buildDir); os.IsNotExist(err) {
+		if err := os.Mkdir(buildDir, os.ModePerm); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	kc1, err := (*multiEnv)[0].GetKubeconfig(t)
+	if err != nil {
+		t.Fatal(err)
+
+	}
+	f, err := os.Create(buildDir + "/kubeconfig1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	if _, err := f.Write(kc1); err != nil {
+		t.Fatal(err)
+	}
+
+	kc2, err := (*multiEnv)[1].GetKubeconfig(t)
+	if err != nil {
+		t.Fatal(err)
+
+	}
+	f, err = os.Create(buildDir + "/kubeconfig2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := f.Write(kc2); err != nil {
+		t.Fatal(err)
+	}
+	ex := RegistrationExecutor{
+		SourceKubeconfig: buildDir + "/kubeconfig1",
+		DestKubeconfig:   buildDir + "/kubeconfig2",
+		SourceContext:    "default-context",
+		DestContext:      "default-context",
+		SourceNamespace:  "source-namespace",
+		DestNamespace:    "dest-namespace",
+		ServiceAccount:   "k8ssandra-operator",
+		Context:          context.TODO(),
+		DestinationName:  "test-destination",
+	}
+	ctx := context.Background()
+
+	require.Eventually(t, func() bool {
+		res := ex.RegisterCluster()
+		switch {
+		case res.IsDone():
+			return true
+		case res.IsError():
+			t.Log(res.GetError())
+			if res.GetError().Error() == "no secret found for service account k8ssandra-operator" {
+				return true
+			}
+		}
+		return false
+	}, time.Second*30, time.Second*5)
+
+	// This relies on a controller that is not running in the envtest.
+
+	desiredSaSecret := &corev1.Secret{}
+	require.NoError(t, client1.Get(context.Background(), client.ObjectKey{Name: "k8ssandra-operator-secret", Namespace: "source-namespace"}, desiredSaSecret))
+	patch := client.MergeFrom(desiredSaSecret.DeepCopy())
+	desiredSaSecret.Data = map[string][]byte{
+		"token":  []byte("test-token"),
+		"ca.crt": []byte("test-ca"),
+	}
+	require.NoError(t, client1.Patch(ctx, desiredSaSecret, patch))
+
+	desiredSa := &corev1.ServiceAccount{}
+	require.NoError(t, client1.Get(
+		context.Background(),
+		client.ObjectKey{Name: "k8ssandra-operator", Namespace: "source-namespace"},
+		desiredSa))
+
+	patch = client.MergeFrom(desiredSa.DeepCopy())
+	desiredSa.Secrets = []corev1.ObjectReference{
+		{
+			Name: "k8ssandra-operator-secret",
+		},
+	}
+	require.NoError(t, client1.Patch(ctx, desiredSa, patch))
+
+	// Continue reconciliation
+
+	require.Eventually(t, func() bool {
+		res := ex.RegisterCluster()
+		switch {
+		case res.IsDone():
+			return true
+		case res.IsError():
+			t.Log(res.GetError())
+			return false
+		}
+		return false
+	}, time.Second*3000, time.Second*5)
```

Comment:
issue: This test should never hang for 3000 seconds. 30s should be maximum, even lower would be better.

Reply:
I'll compromise with you on 300s if you're agreeable. I don't particularly like short timeouts as they historically have caused more flakes than is necessary. A fast test don't become fast just because it times out, a fast test is fast because it passes quickly.

Reply:
Rather fail quickly than hang the GHA run for 3000s for something that's not going to work. It eats a runner and the user has no feedback loop at all. We should have realistic expectations and bail out due to a bug. Otherwise infinite timeout would just as fine if you think we could never hit this - which is not the case.